### PR TITLE
EA Spotlight Item — Add author name and dark mode image support

### PR DESCRIPTION
**Pull request by jpaddison3**
Here's how it looks:

image

We have added the ability to turn on author names. This is a setting on the spotlight item defaulted to false. I've target this at posts where I think it's most useful. You could imagine turning it on for sequences, where it looks only ok:

image

I've also added the ability to add a different image that shows up only during dark mode. That way, you could have an image that works well on a white background with black text above it, and a different image that works well on a black background with white text above it.

The dark mode feature is general to all users of `CloudinaryImage2`.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203797321310446) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by jpaddison3** [comment]
@darkruby501, or maybe @Raemon as the person with the most context.

**Comment by darkruby501** [comment]
Looks fine to me but would get @Raemon's thought on it

File: packages/lesswrong/components/common/CloudinaryImage2.tsx [hunk 1]
```diff
@@ -70,31 +83,62 @@ const CloudinaryImage2 = ({width, height, objectFit, publicId, imgProps, fullWid
...
-    style={imageStyle}
-    className={className}
-  />
+  return <picture>
+    {srcSetFunc && (shouldUseDarkImage === 'maybe' ? <source
+      // Cast is safe for similar reasons to above
+      srcSet={srcSetFunc(darkPublicId!)}
+      media="(min-width: 600px) and (prefers-color-scheme: dark)"
+    /> : <source
+      srcSet={srcSetFunc(publicId)}
```
Comment (Will-Howard) [inline packages/lesswrong/components/common/CloudinaryImage2.tsx:128]: Won't this be the wrong in the case where `shoulduseDarkImage === "yes"`?

File: packages/lesswrong/components/tagging/TagPage.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export const styles = (theme: ThemeType): JssStyles => ({
   },
   imageContainer: {
     width: '100%',
-    '& > img': {
+    '& > picture > img': {
```
Comment (Will-Howard) [inline packages/lesswrong/components/tagging/TagPage.tsx:33]: Hmm I'm a bit concerned about the image -> picture change breaking other places. Have you checked the other places CloudinaryImage2 is used?

**Review by Will-Howard — APPROVED** [review]
ASTLAC (mainly the one comment about the `shouldUseDarkImage === "yes"` case)

File: packages/lesswrong/components/form-components/ImageUpload.tsx [hunk 3]
```diff
@@ -115,7 +121,11 @@ const formPreviewSizeByImageType = {
   spotlightImageId: {
     width: 345,
     height: 234
-  }
+  },
+  spotlightDarkImageId: {
```
Comment (Raemon) [inline packages/lesswrong/components/form-components/ImageUpload.tsx:125]: Do you have an expectation that this class will ever be different from the spotlightImageId class? Seems probably better to just have one class unless there's a specific reason.

Comment (jpaddison3) [reply packages/lesswrong/components/form-components/ImageUpload.tsx:125]: This isn't a class. It's the width and height to pass into CloudinaryImage2 as props, keyed by the image type. I could define a variable,

```ts
const spotlightImageDimensions = {
    width: 345,
    height: 234
}
```

I could use that here:

```ts
  spotlightImageId: spotlightImageDimensions,
  spotlightDarkImageId: spotlightImageDimensions,
```

But that seems to increase the reading complexity to get a small amount DRY, and I don't think it's worth it.

Or below I could do   

```ts
const formPreviewSize = formPreviewSizeByImageType[name === 'spotlightDarkImageId' ? 'spotlightImageId' : name]
```

But I once again don't think it's worth it, for similar reasons.

File: packages/lesswrong/components/tagging/TagPage.tsx (hunk 2 above)
Comment (jpaddison3) [reply packages/lesswrong/components/tagging/TagPage.tsx:33]: I have not checked all uses of CloudinaryImage2, but I've checked for '> img', and the ones I expect to be dicey.

**Pull request merged**